Worm/Worm.js

```diff
@@ -0,0 +1,242 @@
+import { Point } from "../pga2d/objects.js";
+import { WIDTH, HEIGHT } from "../sketchlib/dimensions.js";
+import { fix_mouse_coords } from "../sketchlib/fix_mouse_coords.js";
+import {
+  BeziergonPrimitive,
+  CirclePrimitive,
+  GroupPrimitive,
+  LinePrimitive,
+  PointPrimitive,
+} from "../sketchlib/primitives.js";
+import { Style } from "../sketchlib/Style.js";
+import { Color } from "../sketchlib/Style.js";
+import { draw_primitive } from "../sketchlib/draw_primitive.js";
+import { Motor } from "../pga2d/versors.js";
+import { AnimationChain, Joint } from "./AnimationChain.js";
+import { is_nearly } from "../sketchlib/is_nearly.js";
+import { GooglyEye } from "./GooglyEye.js";
+
+const MIN_BEND_ANGLE = (3 * Math.PI) / 4;
+const ROT90 = Motor.rotation(Point.ORIGIN, Math.PI / 2);
+const ROT45 = Motor.rotation(Point.ORIGIN, Math.PI / 4);
+const SHOW_SPINE = false;
+
+const COLOR_SPINE = new Color(255, 255, 255);
+const SPINE_STYLE = new Style().with_stroke(COLOR_SPINE);
+const CENTER_STYLE = new Style().with_fill(COLOR_SPINE);
+const WORM_STYLE = new Style()
+  .with_fill(new Color(255, 122, 151))
+  .with_stroke(new Color(127, 61, 76));
+
+const WORM_SEGMENTS = 60;
+const WORM_SEGMENT_SEPARATION = 30;
+// speed in pixels per frame
+const WORM_MAX_SPEED = 10;
+const WORM_THICKNESS = 20;
+const WORM_EYE_RADIUS = 0.25 * WORM_THICKNESS;
+const WORM_PUPIL_RADIUS = (3 / 16) * WORM_THICKNESS;
+const WORM_EYE_SEPARATION = 0.5 * WORM_THICKNESS;
+const WORM_NOSE_RADIUS = 1.25 * WORM_THICKNESS;
+const INITIAL_POSITION = Point.point(WIDTH / 2, HEIGHT - 2 * WORM_THICKNESS);
+
+class Worm {
```

Comment:
Let's rename this `AnimatedWorm` and move it to its own file, there's a lot of code here.